mcs/class/System/Mono.Btls/X509CertificateImplBtls.cs

```diff
@@ -285,10 +285,7 @@ void MustFallback ()
 				ThrowIfContextInvalid ();
 				if (issuerName == null) {
 					using (var xname = x509.GetIssuerName ()) {
-						var encoding = xname.GetRawData (false);
-						var canonEncoding = xname.GetRawData (true);
-						var name = MonoBtlsUtils.FormatName (xname, true, ", ", true);
-						issuerName = new X500DistinguishedName (encoding, canonEncoding, name);
+						issuerName = new X500DistinguishedName (xname.GetRawData (false));
 					}
```

Comment:
If we go down this path, then `MonoBtlsUtils.FormatName()` should be removed completely and all its other uses replaced with your new code.   However, we should be careful about comparing different `X500DistinguishedName` and avoid managed parsing.

For BTLS specifically, I was actually planning on replacing that `MonoBtlsUtils.FormatName()` method with a native boringssl based implementation for performance reasons.

Reply:
Unfortunately I don't understand this piece of code well enough. Apparently the CanonicalName is used only in X509ChainMonoImpl. I started with the (wrong) assumption that X509ChainBtlsImpl is used with BTLS and thus it's ok to remove.

Reply:
Yeah, that code is really complicated.  No worries, I can help you with that.

Reply:
This code is gone in my working tree.